[BUG] Error in NPM uninstall command

### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have tested that this issue has not been fixed in the latest [(beta or stable) release](https://github.com/marticliment/WingetUI/releases/).
- [X] I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section for solutions
- [X] This issue is about a bug (if it is not, please use the [correct template](https://github.com/marticliment/WingetUI/issues/new/choose))

### Describe your issue

When you try to uninstall an NPM package it returns an error and the package is not uninstalled.

### Steps to reproduce the issue

_No response_

### WingetUI Log

```text
🔵 Starting <Package: Clock;clock;1.0.2;Npm@global;<PackageManagers.npm.NPMPackageManager object at 0x000002AAF5380C10>;<genericCustomWidgets.TreeWidgetItemWithQAction object at 0x000002AAFC083540>> uninstall with Command ['~\\source\\repos\\personali\\WingetUI\\wingetuiBin\\components\\gsudo.exe', 'npm', 'uninstall', '--global']
['npm ERR! code EUSAGE', 'npm ERR!', 'npm ERR! Remove a package', 'npm ERR!', 'npm ERR! Usage:', 'npm ERR! npm uninstall [<@scope>/]<pkg>...', 'npm ERR!', 'npm ERR! Options:', 'npm ERR! [-S|--save|--no-save|--save-prod|--save-dev|--save-optional|--save-peer|--save-bundle]', 'npm ERR! [-g|--global]', 'npm ERR! [-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]', 'npm ERR! [-ws|--workspaces] [--include-workspace-root] [--install-links]', 'npm ERR!', 'npm ERR! aliases: unlink, remove, rm, r, un', 'npm ERR!', 'npm ERR! Run "npm help uninstall" for more info', 'npm ERR! A complete log of this run can be found in: ~\\AppData\\Local\\npm-cache\\_logs\\2023-08-28T14_40_49_631Z-debug-0.log', '']
```


### Package Managers Logs

```text
Nothing relevant
```


### Relevant information

The problem is in the command given: https://github.com/marticliment/WingetUI/blob/77ea317061bfd1c535ae64ee7d42587ed9fb7a24/wingetui/PackageManagers/npm.py#L276
because the package Id is not specified.


### Screenshots and videos

_No response_

--- a/wingetui/PackageManagers/npm.py
+++ b/wingetui/PackageManagers/npm.py
@@ -294,7 +294,7 @@ class NPMPackageManager(DynamicLoadPackageManager):
     def startUninstallation(self, package: Package, options: InstallationOptions, widget: InstallationWidgetType) -> subprocess.Popen:
         if "@global" in package.Source:
             options.InstallationScope = "Global"
-        Command = [self.EXECUTABLE, "uninstall", ] + self.getParameters(options)
+        Command = [self.EXECUTABLE, "uninstall", package.Id] + self.getParameters(options)
         if options.RunAsAdministrator:
             Command = [GSUDO_EXECUTABLE] + Command
         print(f"🔵 Starting {package} uninstall with Command", Command)

--- a/wingetui/PackageManagers/npm.py
+++ b/wingetui/PackageManagers/npm.py
@@ -100,6 +100,27 @@ class NPMPackageManager(DynamicLoadPackageManager):
                             if not name in self.BLACKLISTED_PACKAGE_NAMES and not id in self.BLACKLISTED_PACKAGE_IDS and not version in self.BLACKLISTED_PACKAGE_VERSIONS and not newVersion in self.BLACKLISTED_PACKAGE_VERSIONS:
                                 packages.append(UpgradablePackage(name, id, version, newVersion, source, Npm))
             globals.PackageManagerOutput += rawoutput
+            p = subprocess.Popen(f"{self.EXECUTABLE} outdated -g", stdout=subprocess.PIPE, stderr=subprocess.STDOUT, stdin=subprocess.PIPE, cwd=os.path.expanduser("~"), env=os.environ.copy(), shell=True)
+            DashesPassed = False
+            rawoutput = "\n\n---------"+self.NAME+"@global"
+            while p.poll() is None:
+                line: str = str(p.stdout.readline().strip(), "utf-8", errors="ignore")
+                rawoutput += "\n"+line
+                if line:
+                    if not DashesPassed:
+                        if "Package" in line:
+                            DashesPassed = True
+                    else:
+                        package = list(filter(None, line.split(" ")))
+                        if len(package) >= 4:
+                            name = formatPackageIdAsName(package[0][1:] if package[0][0] == "@" else package[0]).strip()
+                            id = package[0].strip()
+                            version = package[1].strip()
+                            newVersion = package[3].strip()
+                            source = self.NAME+"@global"
+                            if not name in self.BLACKLISTED_PACKAGE_NAMES and not id in self.BLACKLISTED_PACKAGE_IDS and not version in self.BLACKLISTED_PACKAGE_VERSIONS and not newVersion in self.BLACKLISTED_PACKAGE_VERSIONS:
+                                packages.append(UpgradablePackage(name, id, version, newVersion, source, Npm))
+            globals.PackageManagerOutput += rawoutput
             print(f"🟢 {self.NAME} search for updates finished with {len(packages)} result(s)")
             return packages
         except Exception as e:
@@ -145,8 +166,8 @@ class NPMPackageManager(DynamicLoadPackageManager):
                         package = line.split("@")
                         if len(package) >= 2:
                             idString = '@'.join(package[:-1]).strip()
-                            name = formatPackageIdAsName(idString[1:] if idString[0] == "@" else idString)
-                            id = idString
+                            name = formatPackageIdAsName(idString[1:] if idString[0] == "@" else idString).strip()
+                            id = idString.strip()
                             version = package[-1].strip()
                             if not name in self.BLACKLISTED_PACKAGE_NAMES and not id in self.BLACKLISTED_PACKAGE_IDS and not version in self.BLACKLISTED_PACKAGE_VERSIONS:
                                 packages.append(Package(name, id, version, self.NAME+"@global", Npm))

